### PR TITLE
dcgp-python

### PR DESCRIPTION
**Pull request by bluescarni**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by bluescarni** [comment]
@isuruf would you mind taking a look? I have no idea what's happening here...

The azure builds are giving lots of dependency conflicts when I add pygmo as a dependency, and the appveyor build is failing because of a compilation error in symengine.

We have a CI pipeline for this package which regularly builds it using the conda packages for the dependencies, and we have never experienced any of these issues.

**Comment by bluescarni** [comment]
The only difference in the command line compilation between our CI setup and the conda build, on MSVC, seems to be the use of the -GL flag. But it sounds strange to me that this flag would influence the correct parsing of C++ code?

File: recipes/dcgp-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  name: dcgp-python
+  version: {{ version }}
+
+source:
+  url: https://github.com/darioizzo/dcgp/archive/v{{ version }}.tar.gz
+  sha256: 85c5318cdb514dcd3fc91ba45b9f112bdd4c0a321c311b7b6bd115cf7d03f294
+
+build:
+  number: 0
+  skip: true  # [(win and py<35)]
```
Comment (jan-janssen) [inline recipes/dcgp-python/meta.yaml]: Can you try `win or py<35`

Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: Won't that skip Windows builds altogether?

Comment (jan-janssen) [reply recipes/dcgp-python/meta.yaml]: Yes, I thought that was the intention. Or maybe try just py<35 to fix the build issues.

Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: Ah no, the idea was to have Windows builds only for Python 3 (because for Python 2 the compiler is too old).

Comment (jan-janssen) [reply recipes/dcgp-python/meta.yaml]: But the package conflicts you get currently seem to be limited to Python 2.7.

Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: What selector syntax should I use to disable Python 2.x build on Windows?

Also, are the info in this page still relevant?

https://github.com/conda-forge/staged-recipes/wiki/VC-features

Comment (jan-janssen) [reply recipes/dcgp-python/meta.yaml]: My suggestion is to disable python 2.7 for all systems, just for a test to see if this resolves the issues with the package conflict.

Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: Ok, I'll try. Thanks for the hint.

Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: Looks like the conflicts are still there...

**Comment by bluescarni** [comment]
@isuruf any idea why ``and`` and ``not`` are not accepted in this build but it works fine with a seemingly identical setup (i.e., clang and deps from conda-forge) in our CI?

https://dev.azure.com/darioizzo/dcgp/_build/results?buildId=54

**Comment by bluescarni** [comment]
@isuruf is this looking good to merge?

I worked around the ``and`` issue with a small patch, would you like me to open a PR in symengine to get rid of the ``and``, ``or`` and ``not`` tokens upstream?

File: recipes/dcgp-python/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    - mpir   # [win]
+  run:
+    - boost
+    - pyaudi
+    - pagmo
+    - pygmo
+    - symengine
+    - python
+    - {{ pin_compatible('numpy') }}
+    - ipopt
```
Comment (isuruf) [inline recipes/dcgp-python/meta.yaml]: Is this needed on win?

File: recipes/dcgp-python/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    - numpy  1.14
+    - ipopt  # [unix]
+    - gmp    # [unix]
+    - mpir   # [win]
+  run:
+    - boost
+    - pyaudi
+    - pagmo
+    - pygmo
+    - symengine
```
Comment (isuruf) [inline recipes/dcgp-python/meta.yaml]:
```suggestion
    - {{ pin_compatible("symengine", max_pin="x.x") }}
```

File: recipes/dcgp-python/meta.yaml (hunk 2 above)
Comment (bluescarni) [reply recipes/dcgp-python/meta.yaml]: No, that's a mistake. I'll change it.

**Comment by bluescarni** [comment]
@isuruf I committed the changes, do you think this is looking good now?

File: recipes/dcgp-python/run_test.bat [hunk 4]
```diff
@@ -0,0 +1 @@
+python -c "import dcgpy.test; dcgpy.test.run_test_suite()"
```
Comment (isuruf) [inline recipes/dcgp-python/run_test.bat]: You can make this a `run_test.py` so that it run on both unix and windows

**Comment by bluescarni** [comment]
@isuruf cheers, I'll change that and remove the numpy 1.14 bit in the recipe later tonight.

**Pull request merged**